knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -51,6 +55,45 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch for Kourier resources.
+	manifest, err := kourier.Manifest(ingressNamespace, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+	resources := manifest.Resources()
+
+	gvkToKourier := map[schema.GroupVersionKind][]types.NamespacedName{}
+	for _, u := range resources {
+		gvkToKourier[u.GroupVersionKind()] =
+			append(gvkToKourier[u.GroupVersionKind()], types.NamespacedName{Namespace: u.GetNamespace(), Name: u.GetName()})
+	}
+	manifest, err := kourier.Manifest(ingressNamespace, mgr.GetClient())
+	if err != nil {
+		return err
+	}
+	resources := manifest.Resources()
+
+	for _, u := range resources {
+		err = c.Watch(&source.Kind{Type: &u}, &handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				namespacedNames := gvkToKourier[obj.Object.GetObjectKind().GroupVersionKind()]
+				for _, namespacedName := range namespacedNames {
+					ks, err := common.KnativeServing(mgr.GetClient(), common.NamespaceFromIngressNamespace(ingressNamespace))
```

Comment:
I wanted to call this before the L#76, but it could not List the KnativeServing.

Reply:
We should prolly only call this inside the `if` below, as this will result in one op per resource currently even though we want at most one op.

Reply:
Oh, yes. Thank you. Updated.